File: musheng-tax-system/musheng-business/src/main/java/com/musheng/business/document/entity/DocumentPartyConfig.java
package com.musheng.business.document.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * FBA单据交易方配置实体
 *
 * <p>存储各店铺（siteCode）的交易方信息，同时作为生成器参数类型，
 * 替代原有的 DocumentPartyProperties（application.yml 配置方式）。</p>
 *
 * @author wanhua
 * 10:30 2026年03月22日
 */
@Data
@TableName("t_document_party_config")
@Schema(description = "FBA单据交易方配置")
public class DocumentPartyConfig {

    @TableId(type = IdType.AUTO)
    @Schema(description = "主键")
    private Long id;

    @Schema(description = "站点代码（US/CA/UK/EU），唯一")
    private String siteCode;

    @Schema(description = "买方中文名（必填）")
    private String buyerName;

    @Schema(description = "买方地址")
    private String buyerAddress;

    @Schema(description = "买方电话")
    private String buyerPhone;

    @Schema(description = "买方英文名")
    private String buyerNameEn;

    @Schema(description = "卖方名称（必填）")
    private String sellerName;

    @Schema(description = "卖方地址")
    private String sellerAddress;

    @Schema(description = "卖方电话")
    private String sellerPhone;

    @Schema(description = "供应商名称（必填）")
    private String supplierName;

    @Schema(description = "客户繁体名（必填）")
    private String customerNameTc;

    @Schema(description = "银行账户名")
    private String bankAccountName;

    @Schema(description = "银行账号")
    private String bankAccountNumber;

    @Schema(description = "银行名称")
    private String bankName;

    @Schema(description = "银行地址")
    private String bankAddress;

    @Schema(description = "SWIFT代码")
    private String swiftCode;

    @Schema(description = "创建时间")
    private LocalDateTime createTime;

    @Schema(description = "更新时间")
    private LocalDateTime updateTime;

    @TableLogic
    @Schema(description = "逻辑删除（0=未删除，1=已删除）")
    private Boolean deleted;
}

File: musheng-tax-web/src/api/documentPartyConfig.ts
/**
 * FBA单据交易方配置API接口
 */

import { request } from '@/utils/request'

const BASE_URL = '/api/v1/business/document-party-config'

export interface DocumentPartyConfig {
  id?: number
  siteCode: string
  buyerName: string
  buyerAddress?: string
  buyerPhone?: string
  buyerNameEn?: string
  sellerName: string
  sellerAddress?: string
  sellerPhone?: string
  supplierName: string
  customerNameTc: string
  bankAccountName?: string
  bankAccountNumber?: string
  bankName?: string
  bankAddress?: string
  swiftCode?: string
}

/** 根据站点代码查询配置 */
export function getPartyConfigBySiteCode(siteCode: string) {
  return request.get<DocumentPartyConfig>(`${BASE_URL}/${siteCode}`)
}

/** 查询所有站点配置 */
export function listAllPartyConfigs() {
  return request.get<DocumentPartyConfig[]>(`${BASE_URL}/list`)
}

/** 新增配置 */
export function addPartyConfig(data: DocumentPartyConfig) {
  return request.post<void>(`${BASE_URL}/add`, data)
}

/** 更新配置 */
export function updatePartyConfig(data: DocumentPartyConfig) {
  return request.put<void>(`${BASE_URL}/update`, data)
}

/** 删除配置 */
export function deletePartyConfig(id: number) {
  return request.delete<void>(`${BASE_URL}/${id}`)
}

/** 复制配置到目标站点（目标站点已存在则更新，不存在则新增） */
export function copyPartyConfig(sourceId: number, targetSiteCode: string) {
  return request.post<void>(`${BASE_URL}/copy`, { sourceId, targetSiteCode })
}
